src/test.js

```diff
@@ -733,17 +733,11 @@ function internalStart( test ) {
 }
 
 function numberOfTests( module ) {
-	let count = module.tests.length,
-		modules = [ ...module.childModules ];
 
 	// Do a breadth-first traversal of the child modules
-	while ( modules.length ) {
-		let nextModule =  modules.shift();
-		count += nextModule.tests.length;
-		modules.push( ...nextModule.childModules );
-	}
-
-	return count;
+	return module.childModules.reduce( function( p, childModule ) {
+	    return p + numberOfTests( childModule );
```

Comment:
@leobalter If there are certain expressions we don't want to use that, let me know what those might be and I can try to configure ESLint to restrict them. (Should be pretty easy to restrict arrow functions since they are a separate node type in ESTree.)